contents/staticAnalysis/ESLint.md

```diff
@@ -0,0 +1,111 @@
+# ESLint
+
+Author: [Nicholas Chua](https://github.com/nus-cs3281/2018/blob/master/students/nicholasChua/nicholasChua-Resume.md)
+
+## Overview
+
+[ESLint](https://eslint.org/docs/about/) is an open source JavaScript linter.
+
+Code linting is a type of static analysis that helps find problematic patterns or code that does not follow a certain style guideline. JavaScript, being a loosely-typed, interpreted language, is prone to developer errors and therefore linting is especially important when working with JavaScript.
+
+ESLint is shipped with a wide range of [built-in rules](https://eslint.org/docs/rules/) that can be configured on installation. It also offers flexibility by allowing importing your own custom rules via [plugins](https://eslint.org/docs/developer-guide/working-with-plugins).
+
+## Rules
+ESLint comes shipped with [built-in rules](https://eslint.org/docs/rules/) that can be [configured](#setting-up) to suit your needs.
+
+An example of a rule would be [`brace-style`](https://eslint.org/docs/rules/brace-style). It enforces a specific placement of braces relative to the control statement and body. Brace style is based on preference or organization standards, and ESLint has options to enforce the brace style of your choice. They include:
+
+[One true brace style (1TBS)](https://en.wikipedia.org/wiki/Indentation_style#Variant:_1TBS_(OTBS))
+```js
+if (foo) {
+  bar();
+} else {
+  baz();
+}
+```
+
+[Stroustrup](https://en.wikipedia.org/wiki/Indentation_style#Variant:_Stroustrup)
+```js
+if (foo) {
+  bar();
+}
+else {
+  baz();
+}
+```
+
+[Allman](https://en.wikipedia.org/wiki/Indentation_style#Allman_style)
+```js
+if (foo)
+{
+  bar();
+}
+else
+{
+  baz();
+}
+```
+
+An additional option `allowSingleLine` can be enabled to allow opening and closing braces for a block to be on the same line:
+```js
+if (foo) { bar(); }
+else { baz(); }
+```
+
+If you are unsure about the options of the rule you want to change, you can refer to the comprehensive [rule documentation](https://eslint.org/docs/rules/).
+
+## Setting Up
+
+After [installing with npm](https://eslint.org/docs/user-guide/getting-started#installation-and-usage), ESLint can be configured in [two ways](https://eslint.org/docs/user-guide/configuring#configuring-rules):
+
+**Single configuration file**. In this example, the rule is configured in `.eslintrc.json` and enforces all braces in the **project** to follow Allman style:
+```json
+{
+  "rules": {
+    "brace-style": ["error", "allman"]
+  }
+}
+```
+
+**Configuration comments** inside your file. In this example, the comments enforces all braces in the **file** to follow Allman style:
+```js
+/* eslint brace-style: ["error", "allman"] */
+if (foo === bar)
+{
+  return;
+}

```

Comment:
What's this `error` in the config means? can it be different e.g., `warn`? May be use `error` for one example, and `warn` for the other and also explain how it controls behavior to further illustrate the flexibility?